### ISSUE TITLE
obs-scripting fails to build on Ubuntu 18.04 with python 3.8 (Upstream SWIG issue)

### ISSUE THREAD

**Issue by theofficialgman** [obsproject/obs-studio#6225]
### Operating System Info

Ubuntu 18.04

### OBS Studio Version

Other

### OBS Studio Version (Other)

Master (which is commit https://github.com/obsproject/obs-studio/commit/525c535459811e17ab13eb3fda2d9e9541f0d8ae)

### OBS Studio Log URL

Not relevant

### Expected Behavior

The ability to build with obs-scripting enabled since it has now been enabled by default.

### Current Behavior

Unable to build, errors with the following
```
[ 77%] Built target decklink-output-ui
[ 77%] Building C object UI/frontend-plugins/frontend-tools/CMakeFiles/frontend-tools.dir/frontend-tools.c.o
[ 77%] Building CXX object UI/frontend-plugins/frontend-tools/CMakeFiles/frontend-tools.dir/auto-scene-switcher.cpp.o
[ 77%] Building CXX object UI/frontend-plugins/frontend-tools/CMakeFiles/frontend-tools.dir/output-timer.cpp.o
[ 78%] Building CXX object UI/frontend-plugins/frontend-tools/CMakeFiles/frontend-tools.dir/__/__/combobox-ignorewheel.cpp.o
[ 78%] Building CXX object UI/frontend-plugins/frontend-tools/CMakeFiles/frontend-tools.dir/__/__/double-slider.cpp.o
[ 78%] Building CXX object UI/frontend-plugins/frontend-tools/CMakeFiles/frontend-tools.dir/__/__/horizontal-scroll-area.cpp.o
[ 78%] Building CXX object UI/frontend-plugins/frontend-tools/CMakeFiles/frontend-tools.dir/__/__/properties-view.cpp.o
[ 78%] Building CXX object UI/frontend-plugins/frontend-tools/CMakeFiles/frontend-tools.dir/__/__/qt-wrappers.cpp.o
[ 79%] Building CXX object UI/frontend-plugins/frontend-tools/CMakeFiles/frontend-tools.dir/__/__/spinbox-ignorewheel.cpp.o
[ 79%] Building CXX object UI/frontend-plugins/frontend-tools/CMakeFiles/frontend-tools.dir/__/__/slider-ignorewheel.cpp.o
[ 79%] Building CXX object UI/frontend-plugins/frontend-tools/CMakeFiles/frontend-tools.dir/__/__/vertical-scroll-area.cpp.o
/home/garrett/obs-studio-updates/obs-studio/build/deps/obs-scripting/obspython/CMakeFiles/obspython.dir/obspythonPYTHON_wrap.c: In function ‘SwigPyStaticVar_dealloc’:
/home/garrett/obs-studio-updates/obs-studio/build/deps/obs-scripting/obspython/CMakeFiles/obspython.dir/obspythonPYTHON_wrap.c:3096:3: error: implicit declaration of function ‘_PyObject_GC_UNTRACK’; did you mean ‘_PyObject_GC_IS_TRACKED’? [-Werror=implicit-function-declaration]
   _PyObject_GC_UNTRACK(descr);
   ^~~~~~~~~~~~~~~~~~~~
   _PyObject_GC_IS_TRACKED
cc1: some warnings being treated as errors
deps/obs-scripting/obspython/CMakeFiles/obspython.dir/build.make:75: recipe for target 'deps/obs-scripting/obspython/CMakeFiles/obspython.dir/CMakeFiles/obspython.dir/obspythonPYTHON_wrap.c.o' failed
make[2]: *** [deps/obs-scripting/obspython/CMakeFiles/obspython.dir/CMakeFiles/obspython.dir/obspythonPYTHON_wrap.c.o] Error 1
CMakeFiles/Makefile2:971: recipe for target 'deps/obs-scripting/obspython/CMakeFiles/obspython.dir/all' failed
make[1]: *** [deps/obs-scripting/obspython/CMakeFiles/obspython.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

### Steps to Reproduce

1. `git clone --recursive https://github.com/obsproject/obs-studio.git`
2. `cd obs-studio`
3. `mkdir build`
4. `cd build`
5. `cmake .. -DUNIX_STRUCTURE=1 -DENABLE_PIPEWIRE=OFF -DCMAKE_INSTALL_PREFIX=/usr -DBUILD_BROWSER=OFF -DENABLE_AJA=0`
6. `make`

### Anything else we should know?

I recently built 27.2.3 without any issues.
and yes, with python scripting disabled I am able to build

**Comment by theofficialgman** [comment]
cmake output:
```

**Comment by kkartaltepe** [comment]
We no longer disable functionality if you dont have the required dependencies. It is expected that you will disable all the functionality you dont support explicitly now.

**Comment by WizardCM** [comment]
This is a strange error - we don't define any `_PyObject_GC_**` variables anywhere, so I'm unsure where this conflict could be coming from.

**Comment by WizardCM** [comment]
We recommend running a clean build / full rebuild to ensure any stray bits from an outdated OBS build (or an outdated dependency definition) get replaced properly.

**Comment by theofficialgman** [comment]
> We recommend running a clean build / full rebuild to ensure any stray bits from an outdated OBS build (or an outdated dependency definition) get replaced properly.

@WizardCM @kkartaltepe please re-open, this was a clean build and as seen I have python 3.8. I did exactly as my steps listed. there was no source folder previously.

I have just double verified with another clean pull of the sourcec and build in a totally separate directly and this does happen.

**Comment by kkartaltepe** [comment]
Since python3 is currently appears to be 3.6.5 on ubuntu 18 and your claim python 3.8, im not sure we support building in whatever configuration you seem have managed for yourself. If this occurs in a clean build you likely have an unsupportable system.

**Comment by theofficialgman** [comment]
ubuntu 18.04 has python 3.6.9, python 3.7.5, and python 3.8.0 all available in the official repos. this is not custom and an officially supported install of ubuntu


you can find all of them here:
https://packages.ubuntu.com/bionic/python3.6
https://packages.ubuntu.com/bionic/python3.7
https://packages.ubuntu.com/bionic/python3.8

python apt packages (other dependencies that ubuntu distributes via apt) are based on the python3.6 version. 3.7 and 3.8 are minimal installs

**Comment by theofficialgman** [comment]
also I am just "claming" python3.8 at all, see it in the cmake output I posted

**Comment by theofficialgman** [comment]
anyway... the error originates in an autogenerated file from SWIG during the buildprocess
`obs-studio/build/deps/obs-scripting/obspython/CMakeFiles/obspython.dir/obspythonPYTHON_wrap.c`

it appears this was a longstanding issue in swig and was never fixed until switch 4.0.1
https://git.ligo.org/lscsoft/lalsuite/-/issues/279
https://github.com/swig/swig/issues/1586

**Comment by theofficialgman** [comment]
I've filed a bug report upstream at the ubuntu swig package in hopes they add the patch to bionic swig
https://bugs.launchpad.net/ubuntu/+source/swig/+bug/1966617

No changes here should be necessary

**Comment by Fenrirthviti** [comment]
I'm still on the fence if we keep upstream bugs open just for reference/tracking on our side, but I'll leave this for now. Updated to title to reflect it's an upstream issue.

**Comment by dodgepong** [comment]
I would argue in favor of keeping them open as tracking issues so that they can be closed once we've integrated the upstream change.

**Comment by kkartaltepe** [comment]
there is nothing to integrate, its just a question of if canonical patches their package or not. considering the existing 4 year queue of bugs for swig the likely hood of them incorporating a one off patch (or even responding) seems low. So we would likely leave this open until we drop support for 18.04 or 18.04 end of lifes in 2023 until we close it.

**Comment by theofficialgman** [comment]
Well this issue doesn't block building for 18.04 on most systems. It just happens to be I have python 3.8 installed (alongside the default 3.6.9).
I'll give whoever got notified a week to respond, then I'll email the maintainer who commited the last patch a few months ago.
This is a simple one line change after all

**Comment by RytoEX** [comment]
To clarify, this is an issue with the SWIG package for Ubuntu 18.04 (as noted in https://bugs.launchpad.net/ubuntu/+source/swig/+bug/1966617), which is version 3.0.12. The SWIG PR was merged for SWIG version 4.0.1, which is the version of the package for Ubuntu 20.04. This is nothing for us to change on our end here. If the Ubuntu SWIG package maintainer backports the SWIG fix to the 18.04 package, then this will be fixed.

As we will be dropping Ubuntu 18.04 support, we consider this Issue closed as not planned.
